Enable GPU Process and UI side compositing on macOS

rniwa opened the pull request:
#### 56f0f4203278a87bd40c7509d28c5c1ed61d0568
<pre>
Enable GPU Process and UI side compositing on macOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=254725">https://bugs.webkit.org/show_bug.cgi?id=254725</a>

Reviewed by Simon Fraser.

Enable GPU process for DOM rendering and UI side compositing by default on macOS.

Also add options to disable these two features to run-webkit-tests.

* Source/WTF/wtf/PlatformEnable.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::WebViewImpl):
* Tools/MiniBrowser/mac/SettingsController.m:
(-[SettingsController useUISideCompositing]):
* Tools/Scripts/webkitpy/api_tests/runner.py:
(Runner.command_for_port):
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.cmd_line):
* Tools/WebKitTestRunner/Options.cpp:
(WTR::handleOptionNoRemoteLayerTree):
(WTR::OptionsHandler::OptionsHandler):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::noUseRemoteLayerTree const):
* Tools/WebKitTestRunner/mac/PlatformWebViewMac.mm:
(WTR::PlatformWebView::PlatformWebView):
* Tools/WebKitTestRunner/mac/main.mm:
(setDefaultsToConsistentValuesForTesting):

Canonical link: <a href="https://commits.webkit.org/262629@main">https://commits.webkit.org/262629@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/862a82f2db2e7e4f4ca7c0da096f1b5964d1270d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1400 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1429 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1479 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2295 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1241 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1385 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1490 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1489 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1374 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1413 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1296 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1273 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2142 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/1439 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1304 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1259 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1228 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/1214 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1273 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1295 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2367 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/1358 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1334 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1192 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/1437 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1249 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1266 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/297 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1368 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/1441 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/230 "Built successfully and passed tests") | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/293 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -342,6 +344,9 @@ def parse_args(args):
             help=("Enable all GPU process related features, also set additional expectations and the result report flavor.")),
+        optparse.make_option(
+            "--no-use-gpu-process", action="store_true", default=False,
+            help=("Disable GPU process for DOM rendering.")),
```
Similar to PR #12222, we should consider `arguments.NoAction` here:
```
from webkitcorepy import arguments
...
optparse.make_option(
    '--use-gpu-process', '--no-use-gpu-process',
    dest='use_gpu_process', default=None,
    help='...',
    action=arguments.NoAction,
)
```
Here I think the case for `arguments.NoAction` is much stronger because it looks like we convert these to a feature flag. That would mean `use_gpu_process =True -> UseGPUProcessForDOMRenderingEnabled` and `use_gpu_process =False -> UseGPUProcessForDOMRenderingEnabled=0` and `use_gpu_process =None` would pass nothing (which would have a GPU process enabled in WebKit but disabled in WebKitLegacy if I'm understanding the code correctly?). It does mean that something like `--no-gpu-process` and `--dump-render-tree` is redundant, but I think that's true in the proposed implementation anyways.

rniwa replied:
It doesn't seem to work?

```
% ./Tools/Scripts/run-webkit-tests --debug --no-build --no-show-results fast/scrolling/scroll-animator-basic-events.htm
Traceback (most recent call last):
  File "/Volumes/Data/safari-2/OpenSource/./Tools/Scripts/run-webkit-tests", line 46, in <module>
    sys.exit(main(sys.argv[1:], sys.stdout, sys.stderr))
  File "/Volumes/Data/safari-2/OpenSource/Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py", line 57, in main
    options, args = parse_args(argv)
  File "/Volumes/Data/safari-2/OpenSource/Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py", line 123, in parse_args
    optparse.make_option("--remote-layer-tree", "--no-remote-layer-tree", dest="remote_layer_tree", action=arguments.NoAction, default=None,
  File "/AppleInternal/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/optparse.py", line 581, in __init__
    checker(self)
  File "/AppleInternal/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/optparse.py", line 636, in _check_action
    raise OptionError("invalid action: %r" % self.action, self)
optparse.OptionError: option --remote-layer-tree/--no-remote-layer-tree: invalid action: <class 'webkitcorepy.arguments.NoAction'>
```

JonWBedard replied:
> It doesn't seem to work?
> 
> ...

Forgot that `run-webkit-tests` uses the older `optparse` instead of `argparse`. So never mind, this suggestion won't work.

rniwa replied:
Yeah, it would be a nice cleanup to migrate that over but it's probably outside the scope of this PR.

rniwa committed:
Enable GPU Process and UI side compositing on macOS
https://bugs.webkit.org/show_bug.cgi?id=254725

Reviewed by Simon Fraser.

Enable GPU process for DOM rendering and UI side compositing by default on macOS.

Also add options to disable these two features to run-webkit-tests.

* Source/WTF/wtf/PlatformEnable.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::WebViewImpl):
* Tools/MiniBrowser/mac/SettingsController.m:
(-[SettingsController useUISideCompositing]):
* Tools/Scripts/webkitpy/api_tests/runner.py:
(Runner.command_for_port):
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.cmd_line):
* Tools/WebKitTestRunner/Options.cpp:
(WTR::handleOptionNoRemoteLayerTree):
(WTR::OptionsHandler::OptionsHandler):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::noUseRemoteLayerTree const):
* Tools/WebKitTestRunner/mac/PlatformWebViewMac.mm:
(WTR::PlatformWebView::PlatformWebView):
* Tools/WebKitTestRunner/mac/main.mm:
(setDefaultsToConsistentValuesForTesting):

Canonical link: https://commits.webkit.org/262629@main
